flamedisx/lxe_blocks/quanta_generation.py

```diff
@@ -117,3 +237,83 @@ def annotate_ces(self, d):
         d['quanta_produced_' + bound] = (
                 d['electrons_produced_' + bound]
                 + d['photons_produced_' + bound])
+
+
+def domain_dict_bonus(self, d):
+    # Calculate cross_domains from quanta_produced_noStep and energy
+    mi = self.source._fetch('quanta_produced_noStep_min', data_tensor=d)[:, o]
+    quanta_produced_noStep_domain = mi + tf.range(tf.reduce_max(
+        self.source._fetch('quanta_produced_noStep_dimsizes', data_tensor=d)))
+    energy_domain = self.source.domain('energy', d)
+
+    quanta_produced_noStep = tf.repeat(
+        quanta_produced_noStep_domain[:, :, o],
+        tf.shape(energy_domain)[1],
+        axis=2)
+    energy_noStep = tf.repeat(
+        energy_domain[:, o, :],
+        tf.shape(quanta_produced_noStep_domain)[1],
+        axis=1)
+
+    # Return as domain_dict
+    return dict({'quanta_produced_noStep': quanta_produced_noStep,
+                 'energy_noStep': energy_noStep})
+
+
+def calculate_dimsizes_special(self):
+    d = self.source.data
+
+    # Want electrons and photons to have the same stepping: choose the minimum
+    # of the two for each event
+    quanta_steps = (d['electrons_produced_steps'] <=
+                    d['photons_produced_steps']) * \
+        d['electrons_produced_steps'] \
+        + (d['photons_produced_steps'] <
+           d['electrons_produced_steps']) * d['photons_produced_steps']
+
+    batch_size = self.source.batch_size
+    n_batches = self.source.n_batches
+
+    # Need the electrons/photons steps to be the same within a batch for the
+    # averaging procedure in _compute to work correctly
+    for i in range(n_batches):
+        quanta_steps[i * batch_size: (i + 1) * batch_size + 1] = \
```

Comment:
Super minor comment, but you could assign `i * batch_size` and `(i + 1) * batch_size + 1` to variables rather than computing them twice. To avoid repeating these lines later, you could factor it out into a function (`_unify_batch_steps` or something).